pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -0,0 +1,810 @@
+            serverJobs.createJob(jobOptions, (err, job) => {
+                if (err) {
+                    logger.error('Error in createJob()', err);
+                    serverJobs.failJobSequence(job_sequence_id);
+                    return;
+                }
+
+                job.verbose('Trying to write file');
+                const fullPath = path.join(fileEdit.coursePath, fileEdit.dirName, fileEdit.fileName);
+                fs.writeFile(fullPath, b64DecodeUnicode(fileEdit.editContents), 'utf8', (err) => {
+                    if (err) {
+                        job.fail(err);
+                    } else {
+                        debug(`Wrote file to ${fullPath}`);
+                        job.verbose(`Wrote file to ${fullPath}`);
+                        job.succeed();
+                    }
+                });
+            });
+        };
+
+        const _unstage = function() {
+            debug('create job: unstage');
+            const jobOptions = {
+                course_id: options.course_id,
+                user_id: options.user_id,
+                authn_user_id: options.authn_user_id,
+                job_sequence_id: job_sequence_id,
+                type: 'push_to_remote',
+                description: 'Unstage all changes',
+                command: 'git',
+                arguments: ['reset'],
+                working_directory: fileEdit.coursePath,
+                env: gitEnv,
+                on_success: _add,
+                on_error: _cleanup,
+                no_job_sequence_update: true,
+            };
+            serverJobs.spawnJob(jobOptions);
+        };
+
+        const _add = function() {
+            debug('create job: add');
+            const jobOptions = {
+                course_id: options.course_id,
+                user_id: options.user_id,
+                authn_user_id: options.authn_user_id,
+                job_sequence_id: job_sequence_id,
+                type: 'push_to_remote',
+                description: 'Stage changes to file being edited',
+                command: 'git',
+                arguments: ['add', path.join(fileEdit.dirName, fileEdit.fileName)],
+                working_directory: fileEdit.coursePath,
+                env: gitEnv,
+                on_success: _commit,
+                on_error: _cleanup,
+                no_job_sequence_update: true,
+            };
+            serverJobs.spawnJob(jobOptions);
+        };
+
+        const _commit = function() {
+            debug('create job: commit');
+            const jobOptions = {
+                course_id: options.course_id,
+                user_id: options.user_id,
+                authn_user_id: options.authn_user_id,
+                job_sequence_id: job_sequence_id,
+                type: 'push_to_remote',
+                description: 'Commit changes',
+                command: 'git',
+                arguments: ['commit', '-m', `"in-browser change to ${fileEdit.fileName} by ${fileEdit.uid}"`],
```

Comment:
Would be helpful to add two more arguments here like `'--author', 'Matthew West <mwest@illinois.edu>'` to explicitly set the commit author?

Reply:
Any chance this could be set to the active user information instead of a fixed argument? I'd like to have a good way to audit changes if need be.

Reply:
All of this was addressed.